teuthology/openstack.py

```diff
@@ -0,0 +1,375 @@
+import json
+import logging
+import os
+import paramiko
+import re
+import socket
+import subprocess
+import tempfile
+import teuthology
+
+from .contextutil import safe_while
+from .config import config as teuth_config
+from teuthology import misc
+
+log = logging.getLogger(__name__)
+
+class OpenStack(object):
```

Comment:
I would really like to see docstrings for methods of this class explaining what the args are supposed to be